static/js/settings_bots.js

```diff
@@ -220,6 +223,130 @@ export function update_bot_permissions_ui() {
     }
 }
 
+export function set_up_bot_edit_form(bot) {
+    const html_body = render_edit_bot_form({
+        user_id: bot.user_id,
+        email: bot.email,
+        full_name: bot.full_name,
+        user_role_values: settings_config.user_role_values,
+        disable_role_dropdown: !page_params.is_admin || (bot.is_owner && !page_params.is_owner),
+    });
+
+    let owner_widget;
+    let avatar_widget;
+
+    const bot_type = bot.bot_type.toString();
+    const service = bot_data.get_services(bot.user_id)[0];
+
+    function submit_bot_details() {
+        const role = Number.parseInt($("#bot-role-select").val().trim(), 10);
+        const $full_name = $("#dialog_widget_modal").find("input[name='full_name']");
+        const url = "/json/bots/" + encodeURIComponent(bot.user_id);
+
+        const formData = new FormData();
+        formData.append("csrfmiddlewaretoken", csrf_token);
+        formData.append("full_name", $full_name.val());
+        formData.append("role", JSON.stringify(role));
+
+        if (owner_widget === undefined) {
+            blueslip.error("get_bot_owner_widget not called");
+        }
+        const human_user_id = owner_widget.value();
+        if (human_user_id) {
+            formData.append("bot_owner_id", human_user_id);
+        }
+
+        if (bot_type === OUTGOING_WEBHOOK_BOT_TYPE) {
+            const service_payload_url = $("#edit_service_base_url").val();
+            const service_interface = $("#edit_service_interface :selected").val();
+            formData.append("service_payload_url", JSON.stringify(service_payload_url));
+            formData.append("service_interface", service_interface);
+        } else if (bot_type === EMBEDDED_BOT_TYPE && service !== undefined) {
+            const config_data = {};
+            $("#config_edit_inputbox input").each(function () {
+                config_data[$(this).attr("name")] = $(this).val();
+            });
+            formData.append("config_data", JSON.stringify(config_data));
+        }
+
+        const $file_input = $("#bot-edit-form").find(".edit_bot_avatar_file_input");
+        for (const [i, file] of Array.prototype.entries.call($file_input[0].files)) {
+            formData.append("file-" + i, file);
+        }
+
+        channel.patch({
+            url,
+            data: formData,
+            processData: false,
+            contentType: false,
+            success() {
+                avatar_widget.clear();
+                dialog_widget.close_modal();
+                dialog_widget.hide_dialog_spinner();
```

Comment:
small nit: `close_modal()` removes the modal from the DOM, so calling `hide_dialog_spinner` after is redundant.

Reply:
Ah yes, fixed this.